warehouse/request.py

```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import hashlib
+import hmac
+import secrets
+
+
+def _create_nonce(request):
+    """Generate a cryptographically secure random nonce for the request."""
+    return secrets.token_urlsafe(32)
+
+
+def _create_hashed_domains(request):
+    """Create a list of hashed allowed domains using the request nonce as salt."""
+    allowed_domains = request.registry.settings.get("warehouse.allowed_domains", [])
+
+    if not allowed_domains:
+        return ""
```

Comment:
If I'm reading this correctly, if no domains exist in the config, no detection is applied - is that desirable? Or, should the default then fall back to `warehouse.domain` value so that there's always something in this path of protection?

Reply:
https://github.com/pypi/warehouse/pull/18427/commits/bd9ce90aed39f41afdb8a16de21b085939324c7d